src/wallet/spend.cpp

```diff
@@ -575,6 +584,14 @@ std::optional<SelectionResult> SelectCoins(const CWallet& wallet, CoinsResult& a
         return result;
     }
 
+    // Return early if we cannot cover the target with the wallet's UTXO.
+    // We use the total effective value if we are not subtracting fee from outputs and 'available_coins' contains the data.
+    CAmount available_coins_total_amount = coin_selection_params.m_subtract_fee_outputs ? available_coins.total_amount :
+            (available_coins.total_effective_amount.has_value() ? *available_coins.total_effective_amount : 0);
```

Comment:
We will never use `SelectCoins()` without a feerate. If you end up not having total_effective_amount here, something went terribly wrong. Perhaps just:

```suggestion
    assume(available_coins.total_effective_amount.has_value());
    CAmount available_coins_total_amount = coin_selection_params.m_subtract_fee_outputs ? available_coins.total_amount : *available_coins.total_effective_amount);
```